tutorials/fmc/fmc.c

```diff
@@ -0,0 +1,59 @@
+#include "fmc_config.h"
+#include "libs/adc/api.h"
+#include "libs/timer/api.h"
+#include "tutorials/fmc/can_api.h"
+
+#include <avr/interrupt.h>
+#include <avr/io.h>
+
+volatile bool timer_50hz = true;
+uint16_t adc_value = 0;
+bool high_voltage = false;
+uint8_t high_voltage_counter = 0;
+uint8_t buffer_counter = 0;
+
+void timer0_callback(void) {
+    timer_50hz = true;
+}
+
+void change_duty_cycle(uint8_t torque_request_value) {
+    float duty_cycle = (float)torque_request_value / 255.0;
+    OCR1B = (uint16_t)duty_cycle * 80.0;
+}
+
+int main(void) {
+    sei();
+    timer_init(&timer0_cfg);
+    can_init_fmc();
+    adc_init();
+
+    can_receive_motor_command();
+
+    while (1) {
+        if (timer_50hz) {
+            adc_value = adc_read(adc_pin);
```

Comment:
`adc_pin` isn't a very clear variable name. I usually name ADC pin variables the same way I'd name a GPIO variable (like `TEMPERATURE_SENSE` or something).